gateway/rpc_storage_handler.go

```diff
@@ -875,6 +876,12 @@ func (r *RPCStorageHandler) ProcessKeySpaceChanges(keys []string) {
 				getSessionAndCreate(splitKeys[0], r)
 			} else {
 				log.Info("--> removing cached key: ", key)
+
+				// in case it's an username (basic auth) then generate the token
+				if _, err := base64.StdEncoding.DecodeString(key); err != nil {
```

Comment:
Use `storage.TokenOrg` instead, and check if it is non-empty.  Right now your change will work only for people who have a hashing algorithm set, but will not work with old algo. You can set hash alg function to empty to validate it.